### PR TITLE
Add Help command class and sequence diagrams

### PR DESCRIPTION
**Pull request by appleraincoat**
Add Help Command class and sequence diagrams.

Abit unsure, please lmk if anything is wrong 😢

### CODE REVIEW

**Review by UdhayaShan1 — CHANGES_REQUESTED** [review]
I think class diagram ok, but sequence diagram seems incomplete, after HelpCommand is returned `RealodexParser` should call execute on it which then creates the CommandResult and returns it to HelpCommand

**Review by UdhayaShan1 — APPROVED** [review]
LGTM, all rise

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -343,17 +343,28 @@ Pros: Allows for more flexible searches.
...
+The Help feature provides help to the user (depending on user input) by either giving details on how all commands 
+are used in a new window, or a short description in the main window on how an individual, specified command is used.
 The core components for this feature are:
 - HelpCommand: A command that, when executed, either shows a new window summarising help for all commands, or
 prints the help message in the Main Window for the requested command, depending on user input.
 - HelpCommandParser: Processes the user input to instantiate the HelpCommand object appropriately to perform the
 correct action (the type of help to give, in this case help for all commands).
 
+#### Help Command Architecture
+<puml src="diagrams/help/HelpCommand.puml" width="1000" />
```
Comment (4llysa) [inline docs/DeveloperGuide.md:359]: Thanks for adding the architecture for help.

File: docs/diagrams/help/HelpSequenceDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,65 @@
+@startuml
+!include style.puml
```
Comment (4llysa) [inline docs/diagrams/help/HelpSequenceDiagram.puml:2]: Perhaps this diagram should be included in the DG as well, as right now it is not used.

**Review by 4llysa — APPROVED** [review]
Can add the sequence diagram before merging, or can merge first and we arrange on Friday when we meet

**Pull request merged**